### PR TITLE
Fix/load legend images on request

### PR DESCRIPTION
**Pull request by eSlider**
Done by @philvanbwg.

### CODE REVIEW

**Comment by astroidex** [comment]
* I tested this pull request. Only one file changed mapbender.element.legend.js
* sidepane: it works fine. Only the legend images for services that are active are loaded on start (when legend is in sidepane)
* dialog: is used with dialog the legendimages are loaded when the dialog is opened

* **Disadvantage**: now with this change the image may take some time to show up, when you activate a service (there  is a deplay)

**Solution**
* Maybe it would be good to choose which modus to take - 1. load on start or 2. load on activate
* **with only some WMS in an application** -> modus 1 is fine (pro: image is already loaded)
* **with many services in an application and only some active** -> modus 2 is fine (pro: no delay on start of an application)

File: src/Mapbender/CoreBundle/Resources/public/mapbender.element.legend.js [hunk 1]
```diff
@@ -130,8 +130,14 @@
             } else if(context && options.changed && options.changed.children) {
                 for(layerName in options.changed.children) {
                     var layer = options.changed.children[layerName];
+
                     if(layer.state) {
+
                         if(layer.state.visibility) {
+
+                            if($('li.image[data-id="' + layerName + '"]', context).attr("data-gisurl"))
+                                $('li.image[data-id="' + layerName + '"]', context).html('<img src="' + $('li.image[data-id="' + layerName + '"]', context).attr("data-gisurl") + '">');
```
Comment (atd-schubert) [inline src/Mapbender/CoreBundle/Resources/public/mapbender.element.legend.js]: We should not create HTML from strings, we should prefer creating it like @eSlider [have made it](https://github.com/mapbender/mapbender/blob/2971dc6c735e586da5ede1783ba963145067cb4d/src/Mapbender/CoreBundle/Resources/public/mapbender.element.legend.js#L334)

File: src/Mapbender/CoreBundle/Resources/public/mapbender.element.legend.js [hunk 2]
```diff
@@ -294,23 +300,27 @@
...
+
+            if(layer.visible == " notvisible")
+                return '<li class="ebene' + layer.level + ' ' + layer.visible + ' image" data-id="' + layer.id + '" data-gisurl="' + layer.legend.url + '"></li>';
+            else
+                return '<li class="ebene' + layer.level + ' ' + layer.visible + ' image" data-id="' + layer.id + '"><img src="' + layer.legend.url + '"></img></li>';
         },
         _createTextLine: function(layer, hide){
             return '<li class="ebene' + layer.level + ' ' + layer.visible + ' ' + (hide ? this.hiddeEmpty
-                    : '') + ' text" data-id="' + layer.id + '">' + this.options.noLegend + '</li>';
+                : '') + ' text" data-id="' + layer.id + '">' + this.options.noLegend + '</li>';
```
Comment (atd-schubert) [inline src/Mapbender/CoreBundle/Resources/public/mapbender.element.legend.js]: We should not create HTML from strings, see above...

File: src/Mapbender/CoreBundle/Resources/public/mapbender.element.legend.js [hunk 3]
```diff
@@ -294,23 +300,27 @@
...
-                    ? this.hiddeEmpty : '') + ' subTitle" data-id="' + layer.id + '">' + layer.title + '</li>';
+                ? this.hiddeEmpty : '') + ' subTitle" data-id="' + layer.id + '">' + layer.title + '</li>';
         },
         _createImageLine: function(layer){
-            return '<li class="ebene' + layer.level + ' ' + layer.visible + ' image" data-id="' + layer.id + '"><img src="' + layer.legend.url + '"></img></li>';
+
+            if(layer.visible == " notvisible")
+                return '<li class="ebene' + layer.level + ' ' + layer.visible + ' image" data-id="' + layer.id + '" data-gisurl="' + layer.legend.url + '"></li>';
+            else
+                return '<li class="ebene' + layer.level + ' ' + layer.visible + ' image" data-id="' + layer.id + '"><img src="' + layer.legend.url + '"></img></li>';
```
Comment (atd-schubert) [inline src/Mapbender/CoreBundle/Resources/public/mapbender.element.legend.js]: We should not create HTML from strings, see above...

File: src/Mapbender/CoreBundle/Resources/public/mapbender.element.legend.js [hunk 4]
```diff
@@ -294,23 +300,27 @@
...
         _createTitleLine: function(layer, hide){
             return '<li class="ebene' + layer.level + ' ' + layer.visible + ' ' + this.layerTitle + ' ' + (hide
-                    ? this.hiddeEmpty : '') + ' subTitle" data-id="' + layer.id + '">' + layer.title + '</li>';
+                ? this.hiddeEmpty : '') + ' subTitle" data-id="' + layer.id + '">' + layer.title + '</li>';
         },
         _createImageLine: function(layer){
-            return '<li class="ebene' + layer.level + ' ' + layer.visible + ' image" data-id="' + layer.id + '"><img src="' + layer.legend.url + '"></img></li>';
+
+            if(layer.visible == " notvisible")
+                return '<li class="ebene' + layer.level + ' ' + layer.visible + ' image" data-id="' + layer.id + '" data-gisurl="' + layer.legend.url + '"></li>';
```
Comment (atd-schubert) [inline src/Mapbender/CoreBundle/Resources/public/mapbender.element.legend.js]: We should not create HTML from strings, see above...

File: src/Mapbender/CoreBundle/Resources/public/mapbender.element.legend.js [hunk 5]
```diff
@@ -294,23 +300,27 @@
...
         },
         _createNodeTitleLine: function(layer){
             return '<li class="ebene' + layer.level + ' ' + layer.visible + ' ' + this.grouppedTitle + (!layer.childrenLegend
-                    ? ' notshow'
-                    : '') + ' subTitle" data-id="' + layer.id + '">' + layer.title + '</li>';
+                ? ' notshow'
+                : '') + ' subTitle" data-id="' + layer.id + '">' + layer.title + '</li>';
         },
         _createTitleLine: function(layer, hide){
             return '<li class="ebene' + layer.level + ' ' + layer.visible + ' ' + this.layerTitle + ' ' + (hide
```
Comment (atd-schubert) [inline src/Mapbender/CoreBundle/Resources/public/mapbender.element.legend.js]: We should not create HTML from strings, see above...

File: src/Mapbender/CoreBundle/Resources/public/mapbender.element.legend.js [hunk 6]
```diff
@@ -294,23 +300,27 @@
         },
         _createSourceTitleLine: function(layer){
             return '<li class="ebene' + layer.level + ' ' + this.sourceTitle + (!layer.childrenLegend ? ' notshow'
-                    : '') + ' title" data-sourceid="' + layer.sourceId + '" data-id="' + layer.id + '">' + layer.title + '</li>';
+                : '') + ' title" data-sourceid="' + layer.sourceId + '" data-id="' + layer.id + '">' + layer.title + '</li>';
         },
         _createNodeTitleLine: function(layer){
             return '<li class="ebene' + layer.level + ' ' + layer.visible + ' ' + this.grouppedTitle + (!layer.childrenLegend
```
Comment (atd-schubert) [inline src/Mapbender/CoreBundle/Resources/public/mapbender.element.legend.js]: We should not create HTML from strings, see above...

File: src/Mapbender/CoreBundle/Resources/public/mapbender.element.legend.js (hunk 1 above)
Comment (werrolf) [reply src/Mapbender/CoreBundle/Resources/public/mapbender.element.legend.js]: So the if ($('li.image[data-id ... is the only functional change and everything else on this pull is just whitespace reformatting...?

**Comment by AxxL** [comment]
This PR can be closed because it is done in #606

**Pull request closed without merging**